communication.h

```diff
@@ -9,44 +9,44 @@
 #include <errno.h> // Error number definitions
 #include <sys/types.h>
 
-namespace Communication {
+namespace Communication
+{
+enum Commands {
+    STM32_CMD_GET_COMMANDS = 0x00,
+    STM32_CMD_GET_BL_VERSION = 0x01,
+    STM32_CMD_GET_CHIP_ID = 0x02,
+    STM32_CMD_READ_FLASH = 0x11,
+    STM32_CMD_GO = 0x21,
+    STM32_CMD_WRITE_FLASH = 0x31,
+    STM32_CMD_ERASE_FLASH = 0x43,
+    STM32_CMD_EXTENDED_ERASE_FLASH = 0x44,
+    STM32_CMD_WRITE_UNPROTECT = 0x73,
+    STM32_CMD_INIT = 0x7F,
+    STM32_PARAM_COMPLETE_ERASE = 0xFF
+};
 
-	enum Commands {
-		STM32_CMD_GET_COMMANDS = 0x00,
-		STM32_CMD_GET_BL_VERSION = 0x01,
-		STM32_CMD_GET_CHIP_ID = 0x02,
-		STM32_CMD_READ_FLASH = 0x11,
-		STM32_CMD_GO = 0x21,
-		STM32_CMD_WRITE_FLASH = 0x31,
-		STM32_CMD_ERASE_FLASH = 0x43,
-		STM32_CMD_EXTENDED_ERASE_FLASH = 0x44,
-		STM32_CMD_WRITE_UNPROTECT = 0x73,
-		STM32_CMD_INIT = 0x7F,
-		STM32_PARAM_COMPLETE_ERASE = 0xFF
-	};
+enum Results {
```

Comment:
Deine Results haben mehrere Probleme. Zum einen verwendest du die Results nicht in der Communication Klasse. Die Namen der Results lassen schon auf eine Verwendung in der Bootloader Klasse schließen. Daher ist dort der geeignete Platz. Weiterhin muss du dir fuer deine Communication Klasse eine Strategie fuer Rueckgabewerte ueberlegen. Beispielsweise ist in der Linux Welt die Rueckgabe von 0 bei OK verbreitet, und negative Zahlen fuer Fehler. Stellenweise willst du aber auch gesendete Zeichen als Positive Zahl zurueckgeben. Wenn du nun ein size_t angibst, kannst du keine negativen Zeichen fuer Fehler verwenden. Also solltest du entweder mit '''int''' arbeiten, oder mit true false, oder eine ganz eigne Strategie verwenden.